src/main.ts

```diff
@@ -21,21 +21,26 @@ import { ConsumeEventStreamRecord } from './domain/ConsumeEventStreamRecordRepos
 import { ConsumeEventStreamUseCase } from './useCases/ConsumeEventStreamUseCase';
 import { GetNotificationDocumentMetadataUseCase } from './useCases/GetNotificationDocumentMetadataUseCase';
 import { GetNotificationDocumentMetadataRecord } from './domain/GetNotificationDocumentMetadataRepository';
+import { GetPaymentNotificationMetadataUseCase } from './useCases/GetPaymentNotificationMetadataUseCase';
+import { GetPaymentNotificationMetadataRecord } from './domain/GetPaymentNotificationMetadataRepository';
 
 pipe(
   parseConfig(process.env),
   E.map((config) => {
     const logger = makeLogger();
     const mkRepository = inMemory.makeRepository(logger);
     /* put here the driven adapters (e.g.: Repositories ) */
-    const preLoadRecordRepository = mkRepository<PreLoadRecord>([]);
-    const uploadToS3RecordRepository = mkRepository<UploadToS3Record>([]);
-    const newNotificationRepository = mkRepository<NewNotificationRecord>([]);
-    const createEventStreamRecordRepository = mkRepository<CreateEventStreamRecord>([]);
-    const checkNotificationStatusRepository = mkRepository<CheckNotificationStatusRecord>([]);
-    const getNotificationDetailRepository = mkRepository<GetNotificationDetailRecord>([]);
-    const consumeEventStreamRepository = mkRepository<ConsumeEventStreamRecord>([]);
+    const preLoadRecordRepository = inMemory.makeRepository(logger)<PreLoadRecord>([]);
+    const uploadToS3RecordRepository = inMemory.makeRepository(logger)<UploadToS3Record>([]);
+    const newNotificationRepository = inMemory.makeRepository(logger)<NewNotificationRecord>([]);
+    const createEventStreamRecordRepository = inMemory.makeRepository(logger)<CreateEventStreamRecord>([]);
+    const checkNotificationStatusRepository = inMemory.makeRepository(logger)<CheckNotificationStatusRecord>([]);
+    const getNotificationDetailRepository = inMemory.makeRepository(logger)<GetNotificationDetailRecord>([]);
+    const consumeEventStreamRepository = inMemory.makeRepository(logger)<ConsumeEventStreamRecord>([]);
     const getNotificationDocumentMetadataRecordRepository = mkRepository<GetNotificationDocumentMetadataRecord>([]);
+    const getPaymentNotificationMetadataRepository = inMemory.makeRepository(
+      logger
+    )<GetPaymentNotificationMetadataRecord>([]);
```

Comment:
I've replaced `inMemory.makeRepository(logger)` with `mkRepository` which is a partially applied function

```suggestion
    const preLoadRecordRepository = mkRepository<PreLoadRecord>([]);
    const uploadToS3RecordRepository = mkRepository<UploadToS3Record>([]);
    const newNotificationRepository = mkRepository<NewNotificationRecord>([]);
    const createEventStreamRecordRepository = mkRepository<CreateEventStreamRecord>([]);
    const checkNotificationStatusRepository = mkRepository<CheckNotificationStatusRecord>([]);
    const getNotificationDetailRepository = mkRepository<GetNotificationDetailRecord>([]);
    const consumeEventStreamRepository = mkRepository<ConsumeEventStreamRecord>([]);
    const getNotificationDocumentMetadataRecordRepository = mkRepository<GetNotificationDocumentMetadataRecord>([]);
    const getPaymentNotificationMetadataRepository = mkRepository<GetPaymentNotificationMetadataRecord>([]);
```

Reply:
My bad! I kept the wrong version during the merge! I'm going to fix it

Reply:
Changed it!